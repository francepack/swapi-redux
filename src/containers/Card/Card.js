import React from 'react'
import PropTypes from 'prop-types'

const Card = (props) => {
  let star;
  let renderCard;
  star =
  <div className='star' onClick={props.toggleFav}>
    {props.favorite &&
      <img src="https://img.icons8.com/windows/32/FCB001/filled-star.png" alt='active-star' />
    }
    {!props.favorite &&
      <img src="https://img.icons8.com/windows/32/0F3D56/filled-star.png" alt='inactive-star' />
    }
  </div>  

  renderCard =
    <div className='card-content'>
      <header>
        <h2>{props.name}</h2>
        <div>{star}</div>
      </header>
      <article>
        {props.species && 
          <p>Species: <span>{props.species}</span></p>
        }
        {props.homeworld && 
          <p>Home: <span>{props.homeworld}</span></p>
        }
        {props.population && 
          <p>Home pop: <span>{props.population}</span></p>
        }
        {props.terrain && 
          <p>Terrain: <span>{props.terrain}</span></p>
        }
        {props.climate && 
          <p>Climate: <span>{props.climate}</span></p>
        }
        {props.model && 
          <p>Model: <span>{props.model}</span></p>
        }
        {props.class && 
          <p>Class: <span>{props.class}</span></p>
        }
        {props.passengers && 
          <p>Passengers: <span>{props.passengers} people</span></p>
        }
        {props.residents && 
          <div>
            <p className="resident-head">Residents of Note</p>
            <ul>
              {!props.residents.length && 
                <li>None</li>
              }
              {props.residents.map(resident => {
                return <li key={resident}>{resident}</li>
              })}
            </ul>
          </div>
        }
      </article>
    </div>

  return(
    <div className='card'>
      {renderCard} 
    </div>
  )
}


Card.propTypes = {
  person: PropTypes.object,
  planet: PropTypes.object,
  vehicle: PropTypes.object
}

export default Card