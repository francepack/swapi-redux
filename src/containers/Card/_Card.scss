.card {
  background-color: #e5dbb3;
  border: 4px solid $lightTan;
  border-radius: 15px;
  width: 31.2%;
  margin: 30px 1% 0 1%;
}

.card-content span {
  font-family: 'Roboto Mono', monospace;
}

.card-content header {
  align-items: center;
  color: $midnightGreen;
  display: flex;
  justify-content: space-between;
  position: relative;
}

.card-content header h2 {
  font-family: 'Bangers', cursive;
  letter-spacing: 2px;
  margin-left: 20px;
}

.card-content article {
  color: $midnightGreen;
}

.resident-head {
  font-style: italic;
  text-decoration: underline;
}

.card-content ul {
  padding-left: 0;
}

.card-content li {
  list-style: none;
}

.star {
  margin-right: 10px;
  &:hover {
    cursor: pointer;
  }
}