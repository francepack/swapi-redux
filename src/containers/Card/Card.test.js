import React from 'react';
import { enzyme, shallow } from 'enzyme';
import Card from './Card';

describe('card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card />
    )
  });  
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  });
});