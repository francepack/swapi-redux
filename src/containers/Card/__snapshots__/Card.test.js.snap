// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-content"
      >
        <header>
          <h2 />
          <div>
            <div
              className="star"
            >
              <img
                alt="inactive-star"
                src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
              />
            </div>
          </div>
        </header>
        <article />
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h2 />
            <div>
              <div
                className="star"
              >
                <img
                  alt="inactive-star"
                  src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                />
              </div>
            </div>
          </header>,
          <article />,
        ],
        "className": "card-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <div>
                <div
                  className="star"
                >
                  <img
                    alt="inactive-star"
                    src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                  />
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="star"
                >
                  <img
                    alt="inactive-star"
                    src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                  />
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    <img
                      alt="inactive-star"
                      src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                    />,
                  ],
                  "className": "star",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "inactive-star",
                      "src": "https://img.icons8.com/windows/32/0F3D56/filled-star.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-content"
        >
          <header>
            <h2 />
            <div>
              <div
                className="star"
              >
                <img
                  alt="inactive-star"
                  src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                />
              </div>
            </div>
          </header>
          <article />
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <h2 />
              <div>
                <div
                  className="star"
                >
                  <img
                    alt="inactive-star"
                    src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                  />
                </div>
              </div>
            </header>,
            <article />,
          ],
          "className": "card-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2 />,
                <div>
                  <div
                    className="star"
                  >
                    <img
                      alt="inactive-star"
                      src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                    />
                  </div>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="star"
                  >
                    <img
                      alt="inactive-star"
                      src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                    />
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      <img
                        alt="inactive-star"
                        src="https://img.icons8.com/windows/32/0F3D56/filled-star.png"
                      />,
                    ],
                    "className": "star",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "inactive-star",
                        "src": "https://img.icons8.com/windows/32/0F3D56/filled-star.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            "type": "article",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
